### PR TITLE
Add extra functions to eql.utils

### PR DESCRIPTION
**Pull request by rw-access**
## Issues
None

## Details
Added a few utility functions. One use case was categorizing and building stats on our existing analytics to get a sense of how often each feature is used. This can help us get a sense of our expectations for EQL in Elasticsearch.

### CODE REVIEW

File: eql/utils.py [hunk 1]
```diff
@@ -243,6 +243,30 @@ def is_stateful(query):
...
+    :param ast.PipedQuery|ast.Analytic query: The parsed query AST to analyze
+    :rtype: str
+    """
+    from . import ast  # noqa: E402
+
+    if isinstance(query, ast.EqlAnalytic):
+        query = query.query
+
+    elif not isinstance(query, ast.PipedQuery):
+        raise TypeError("unsupported type {} to eql.utils.get_query_type. Expected {}".format(type(query), ast.EqlNode))
```
Comment (brokensound77) [inline eql/utils.py:258]: Can you create a custom exception for this which inherits the base eql exception and `TypeError` to catch this easier

Comment (rw-access) [reply eql/utils.py:258]: I think it's okay in these utility functions. It'd be the same exception as if you called with one arg or too many args. It's more of a curse of dynamic typing than anything else. So it's not an EQL-specific error, but a "you used python wrong" error

Comment (rw-access) [reply eql/utils.py:258]: I think in `match_kv` that should probably be an EqlError instead

Comment (rw-access) [reply eql/utils.py:258]: Or rather, use `parse_field` instead

**Pull request merged**